File: Java/4.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner in = new Scanner(System.in);
        System.out.println("Enter candy count(kg): ");
        int candy_count = in.nextInt();
        System.out.println("Enter candy price: ");
        int candy_price = in.nextInt();

        System.out.println("Enter how much do you need: ");
        int candy_count_user = in.nextInt();
        int candy_price_user = (candy_price * candy_count_user) / candy_count;
        System.out.println("It will be cost: " + candy_price_user);

        System.out.println("How much do you have? ");
        int money_user = in.nextInt();
        int candy_user = (candy_count * money_user) / candy_price;
        System.out.println("You can buy: " + candy_user);
    }
}

File: Python/4.py
def main():
    candies_user_mass = int(input("Enter candies mass: "))
    candies_user_price = int(input("Enter candies price: "))
    candies_price = candies_user_price / candies_user_mass

    user_candies_count = int(input("Enter how much do you need(kg): "))
    user_candies_price = candies_price * user_candies_count
    print("You need to pay: ", user_candies_price)

    user_candies_money = int(input("Enter how much money do you have: "))
    user_candies_mass = user_candies_money / candies_price
    print("You can buy: ", user_candies_mass)

if __name__ == "__main__":
    main()